app/Http/Controllers/GroupController.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Judite\Models\Group;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Http\Request;
+use App\Judite\Models\Course;
+use App\Judite\Models\User;
+use Maatwebsite\Excel\Facades\Excel;
+use Illuminate\Support\Facades\Mail;
+use App\Judite\Models\Student;
+
+class GroupController extends Controller
+{
+    /**
+     * Create a new controller instance.
+     */
+    public function __construct()
+    {
+        $this->middleware('auth');
+        #$this->middleware('can.student');
+        #$this->middleware('can.group')->only(['create', 'store']);
+        $this->middleware('student.verified');
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\View\View
+     */
+    public function index()
+    {
+
+        $student = Auth::student();
+
+        $pendingGroups = DB::transaction(function () use ($student) {
+            return $student->pendingGroups()->get();
+        });
+
+        $confirmedGroups = DB::transaction(function () use ($student) {
+            return $student->confirmedGroups()->get();
+        });
+
+        return view('groups.index', compact('student', 'pendingGroups', 'confirmedGroups'));
+    }
+
+    public function adminIndex()
+    {
+
+        $groups = DB::table('groups')->get();
+
+        $courses = DB::transaction(function () {
+            return Course::orderedList()->get();
+        });
+
+        return view('groups.admin', compact('groups', 'courses'));
+    }
+
+    public function create()
+    {
+        $student = Auth::student();
+
+        $courses = $student->getCoursesWithoutGroup();
+
+        return view('groups.create', compact('courses'));
+    }
+
+    /**
+     * 
+     */
+    public function store(Request $request) {
+        
+        $atributtes = $request->validate([
+            'email' => 'required',
+            'course_id' => 'required'
```

Comment:
Please validate that this course id exists, inside the transaction enviromnent 😄